### PR TITLE
Generate camelCase function names from non-identifier operation ids

### PR DESCRIPTION
**Pull request by christianhelle**
## What changed

Operation ids that are not valid Zig identifiers now generate camelCase declarations instead of quoted ones:

```
@"repos/list-pull-requests-associated-with-commit"  ->  reposListPullRequestsAssociatedWithCommit
@"pull-request-stacks/create"                       ->  pullRequestStacksCreate
```

Ids that are already valid identifiers (`addPet`, `get_pet_by_id`) pass through untouched, so specs with hand-written ids generate exactly what they did before.

## Why

The GitHub REST API spec (`openapi/v3.0/github.json`) uses ids such as `repos/list-pull-requests-associated-with-commit`, which previously produced `pub fn @"repos/list-pull-requests-associated-with-commit"(...)`. Valid Zig, but awkward to call and to read.

## How

`toZigMethodNameAlloc` in `src/generators/unified/ident_utils.zig` does the conversion. `UnifiedApiGenerator.reserveOperationNames` assigns one name per operation up front, in path/method order so the output is stable regardless of hash-map iteration order, and caches it for the lifetime of the generate call. Every site that read `operation.operationId` now goes through that cache, so the flat function, its `Raw`/`Result`/`Streaming` variants, the options struct, resource wrappers and tag clients all agree on one name.

Edge cases covered by unit tests on the conversion itself: a name that would start with a digit is prefixed (`2fa/enable` -> `_2faEnable`); an id with nothing usable in it (`///`) falls back to the original so the existing escaping still applies; a reserved word (`if`) is left alone and stays `@"if"`.

## Collisions this exposed, and how they are resolved

- `get-pet` and `getPet` collapse onto the same camelCase name. The later one in path order becomes `getPet_`.
- GitHub declares both `markdown/render` and `markdown/render-raw`, and the first already owns `markdownRenderRaw` through its `Raw` variant. Each operation now reserves its whole family of function names (base, `Raw`, `Result`, `Stream`, `StreamEvents`, `Streaming`), not just the base. The options struct type is deliberately left out of that family, since `generateOptionsType` already disambiguates it against every top-level declaration.

## Latent bug fixed along the way

Tag clients decided whether to route a delegation through a `_`-prefixed alias by comparing the *operation id* against the name the id would derive, rather than the *final* method name after collision suffixing. With camelCasing in play this produced `pub fn init_(...) { return init(...); }`, where `init` resolves to the tag client struct's own `init` constructor. The predicate now asks whether the flat function name is shadowed by anything declared inside the struct (a sibling method name, or one of `init`/`deinit`/`client`), which reproduces every previously expected alias and fixes the new cases.

## Notes for reviewers

- **This is a breaking change for generated code.** Any downstream code calling the previously quoted functions will need updating.
- `examples/github/client.zig` and `generated/openai.zig` are regenerated in their own commit, so the naming change is easy to read separately from the generator change.
- Three existing tests changed expectations. `multiple_clients_tests` and `parameters_as_struct_tests` assert the new names for their deliberate-collision fixtures. In `sse_cancel_tests`, the fixture's second operation id was `chat-completion-streaming`, which relied on camelCase and PascalCase derivations of one id differing; that is no longer true by design, so it was renamed to `chat-completion-events` to restore the discriminator the test depends on.

## Verification

- `zig build test`: 10/10 steps succeeded, 468 passed / 6 skipped / 0 failed.
- The full GitHub spec (~900 operations) generates and passes `zig ast-check`. A whole-program compile of it produces two format-string errors, which I confirmed against a baseline build of `main` are pre-existing and unrelated to naming.
- Built with nano commits and TDD; each behavior has a failing test before its implementation.

## Summary by CodeRabbit

* **Documentation**
  * Documented how generated endpoint names are derived from OpenAPI operation IDs, including casing and collision handling.

* **Improvements**
  * Improved generated API method naming for invalid, missing, or conflicting operation IDs.
  * Added deterministic suffixes and aliases to prevent naming collisions.
  * Improved naming of generated options types and client methods.
  * Added support for resolving names consistently across generated endpoint variants.

* **Tests**
  * Added coverage for naming conversion, missing operation IDs, and collision scenarios.

### CODE REVIEW

File: src/generators/unified/api_generator/tag_clients.zig [hunk 1]
```diff
@@ -436,23 +465,11 @@ pub fn tagClientMethodNameAlloc(self: *UnifiedApiGenerator, operation_id: []cons
     return method;
 }
 
-pub fn generateTagClientMethod(self: *UnifiedApiGenerator, struct_name: []const u8, method_name: []const u8, op_ref: OperationRef) !void {
+pub fn generateTagClientMethod(self: *UnifiedApiGenerator, struct_name: []const u8, method_name: []const u8, op_ref: OperationRef, needs_alias: bool) !void {
     const operation = op_ref.operation;
-    const op_id = operation.operationId;
+    const op_id = try self.operationNameOf(operation);
```
[the remark this answers is not in the document]
Comment (christianhelle) [reply src/generators/unified/api_generator/tag_clients.zig]: Confirmed and fixed in 0d88ca1 — I reproduced it as a real `error: ambiguous reference`. The events delegation now uses its own `needs_events_alias` flag (an operation's own members never shadow this helper, since the method is `{name}StreamEvents` while the helper is `{name}StreamingEvents`, so reusing `needs_alias` would have over-aliased and changed unrelated output), and `_{op}StreamingEvents` is emitted only when some sibling method actually carries that name.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/tests/multiple_clients_tests.zig [hunk 2]
```diff
@@ -429,12 +429,14 @@ test "multiple-clients PerTag dedupes struct names, reserved methods, and method
     try std.testing.expect(std.mem.indexOf(u8, code, "const _ifRaw = ifRaw;") != null);
     try std.testing.expect(std.mem.indexOf(u8, code, "return _ifRaw(self.client, requestBody);") != null);
 
-    // Method collision: get-pet and getPet both sanitize to getPet → getPet, getPet_.
+    // Method collision: get-pet and getPet both camel case to getPet, so the
+    // flat functions become getPet and getPet_ and the methods follow suit.
```
[the remark this answers is not in the document]
Comment (christianhelle) [reply src/tests/multiple_clients_tests.zig:436]: Fixed in d0c7387 — this test now uses `test_utils.createTestAllocator()` with the `gpa.deinit() == .ok` assertion, per the guideline in `.github/copilot-instructions.md`. Scoped to the one test this PR modified; the other four in the file still use `std.testing.allocator` and are untouched.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/generators/unified/api_generator.zig [hunk 3]
```diff
@@ -94,12 +103,105 @@ pub const UnifiedApiGenerator = struct {
...
+    }
+
+    /// Every function one operation name gives rise to. An operation named
+    /// `foo` declares `foo` itself plus `fooRaw`, `fooResult` and the rest, so
+    /// two operations conflict when any of these names coincide -- GitHub's
+    /// `markdown/render` and `markdown/render-raw` are one such pair. The
+    /// options struct type is left out: it is disambiguated against every
+    /// top-level declaration by generateOptionsType, so an operation named
+    /// `fooOptions` need not move aside for one named `foo`.
+    const declaration_suffixes = [_][]const u8{ "", "Raw", "Result", "Stream", "StreamEvents", "Streaming" };
```
[the remark this answers is not in the document]
Comment (christianhelle) [reply src/generators/unified/api_generator.zig]: Correct, and fixed in 887c085 — the emitted helpers are `{op}Streaming` and `{op}StreamingEvents`, so `StreamingEvents` is now reserved. I also dropped `Stream`/`StreamEvents` from the list: those name wrapper and tag-client methods, which live inside a struct rather than at file scope, so reserving them only caused spurious renames.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
